### ISSUE TITLE
Bumping up required Java version to 11?

### ISSUE THREAD

**Issue by skyhirider** [HtmlUnit/htmlunit#617]
Now that Selenium [will require java 11](https://www.selenium.dev/blog/2023/java-8-support/), have you considered bumping up the version for HtmlUnit?

I am wondering if there is anything practical there that could help out while you are building your library.

**Comment by rbri** [comment]
At the moment i have no plans to switch to JDK 11.

**Comment by skyhirider** [comment]
Makes sense. I saw another issue where you talked about switching to httpclient 5. Once you eventually decide to upgrade, java 11 has its own new httpclient which could replace the one you are already using.

**Comment by rschwietzke** [comment]
Would favor JDK 11 as minimal requirement as well. Just to leave some legacy behind us as well as benefit from language and library updates. Maybe we should run a poll to see if anyone still needs JDK8.

**Comment by rbri** [comment]
Current situation:
Development and running the test suite requires JDK 11.
The release is done using JDK8 and i like to stay with this for some time.

Next step is to set up an working android development environment to be able to test the htmlunit-android build with various android version.
And there is something more to do regarding css-parser and javacc to fix another major problem with the androind support.

And then i will start thinking about moving forward....

**Comment by rbri** [comment]
@RussiaVk what is the reason for your down vote

**Comment by RussiaVk** [comment]
> @RussiaVk what is the reason for your down vote

Sorry,I just think that maintaining compatibility with older versions of Java is friendlier.

**Comment by skyhirider** [comment]
Eventually old dependencies need to be updated to provide security fixes, extend support and allow better workflow with new syntax as the language evolves and expands over time.

You can't stay on old java forever. Hopefully with the spread of AIs it will become much easier to upgrade old code.

Plus, if I remember correctly oracle is charging for updates to old versions unless you're using an open variant, so there is cost involved as well.

**Comment by rbri** [comment]
5.0.0 is now on jdk 17
